### PR TITLE
Use the new omni apis for client

### PR DESCRIPTION
**Pull request by ericanderson**
Blocks on #320

Broken into two commits to make review easier. Its a pretty small change over all but it does generate a lot of code.

### CODE REVIEW

File: packages/platform-sdk-generator/src/model/Model.ts [hunk 1]
```diff
@@ -80,8 +82,11 @@ export class Model {
           mapObjectValues(dt.object.properties, p => this.getType(p.type)),
         );
       case "asyncOperation":
-      case "binary":
+        return new BuiltinType({ type: "any", any: {} });
         throw new Error("Method not implemented.");
```
Comment (ssanjay1) [inline packages/platform-sdk-generator/src/model/Model.ts:86]: Do you still want this throw here?

File: packages/platform-sdk-generator/src/model/Resource.ts [hunk 2]
```diff
@@ -18,6 +18,6 @@ import type { Component } from "./Component.js";
 import type { StaticOperation } from "./StaticOperation.js";
 
 export interface Resource {
-  component: Component;
+  component: string;
```
Comment (ssanjay1) [inline packages/platform-sdk-generator/src/model/Resource.ts:21]: Why is this now typed as a string?

**Review by ssanjay1 — APPROVED** [review]
Approved, couple of quick questions

File: packages/platform-sdk-generator/src/model/Resource.ts (hunk 2 above)
Comment (ericanderson) [reply packages/platform-sdk-generator/src/model/Resource.ts:21]: You arent required to have a real component it turns out. I was only using this for the name anyway.

**Pull request merged**